### ISSUE TITLE
MediaNotFoundException is not catched in media redirect

### ISSUE THREAD

**Issue by Jupi007** [sulu/sulu#6924]
| Q | A
| --- | ---
| Sulu Version | 2.5.4
| PHP Version | 8.1.11

# Actual Behavior

Trying to access to the `sulu_media.redirect` route with a non existent media id, throw a `MediaNotFoundException` which is not caught by the controller.

# Expected Behavior

I could be wrong, but I think this error should be caught to throw a `NotFoundHttpException` instead.
This mean that the http code should be 404 and not 500.

# Steps to Reproduce

In a raw way: access to `/admin/media/redirect/media/{id}?locale={locale}`, with a non existent media id.

In fact, I'm using a monolog email sender to report errors, and sometimes, when someone remove a media, this exception is throw shortly after, and I'm receiving an email (I received many ones each time for a different media which was recently removed).
So I'm sorry, but I don't know the exact way to reproduce this error _in Sulu admin itself_. I'm investigating, but still not very familiar with Sulu, so I preferred to report this problem as it in priority.

<del>
# Possible Solutions

With this simple patch, the 500 error is converted in a 404 one:

```patch
-        /** @var Media $media */
-        $media = $this->mediaManager->getById($id, $locale);
+        try {
+            /** @var Media $media */
+            $media = $this->mediaManager->getById($id, $locale);
+        } catch (MediaNotFoundException $e) {
+            throw new NotFoundHttpException($e->getMessage());
+        }
```

Here: https://github.com/sulu/sulu/blob/8ed3425211e8b72b4ac7ecc5239842408927fc42/src/Sulu/Bundle/MediaBundle/Controller/MediaRedirectController.php#L51-L52

But this is surly not the right way to fix that, maybe the same thing in the manager?
</del>

**Comment by mamazu** [comment]
I think we should aim for a more general solution to convert all MediaNotFoundExceptions to 404 pages. The simplest solution there would be to add a configuration in Symfony to output:
https://lindevs.com/define-http-status-code-of-exception-in-symfony/

or if that doesn't work a event listener for `kernel.exception` would also work.

**Comment by Jupi007** [comment]
@mamazu Your solution seems the cleanest one possible.
I just tested to add a framework configuration extension to the media bundle like you said, and it works.

Do you want to create a PR, or can I do it?

**Comment by mamazu** [comment]
Sure, you can do that. I am not yet clear what the easiest way would be. But if you already have an idea, go ahead.

**Comment by Jupi007** [comment]
Problem: `framework.exception` is only available up to [Sf 5.4](https://symfony.com/doc/5.4/reference/configuration/framework.html), but Sulu 2.4 still supports Sf 4.4.
So I need to work on a `kernel.exception` listener instead.

**Comment by mamazu** [comment]
Ah yeah, I mean you could add a note to it referring to this issue so that it can be migrated in the future.

**Comment by Jupi007** [comment]
You're right, Even better, what do you think of this solution: I can create a PR with a listener for sulu 2.4, and another one using the `framework.exception` feature for sulu 2.5, which the minimal Sf version is 5.4.

Would it be a good idea? I prefer to ask, because I know that the 2.4 branch is merged into the 2.5 one at each release, and I don't want to create unwanted conflicts.

**Comment by mamazu** [comment]
I don't know what the policy for this is in sulu.

@alexander-schranz Any opinion on it?

**Comment by alexander-schranz** [comment]
Sorry for the late response here. I would not create a general listener. A `MediaNotFoundException` does in a business case not mean that it is 404 error, if it is thrown as part of another entity. It could if required also be a 500 error when something is not keep in mind. The 404 would not be tracked then by Sentry which could hide some implementation errors.

So I recommend to catch the MediaNotFoundException specific inside the Redirect Controller and not in a general listener.

**Comment by Jupi007** [comment]
Okay, thanks for the answer @alexander-schranz 👍
I will do this so.

**Comment by Jupi007** [comment]
Closed by #7005
